lib/ActionField.js

```diff
@@ -30,7 +30,7 @@ class ActionField extends Immutable.Record({
 			map.set('name', json.name || map.name);
 			map.set('title', json.title || map.title);
 			map.set('type', json.type || map.type);
-			map.set('value', json.value || map.value);
+			map.set('value', (json.value === false || json.value === 0) ? json.value : json.value || map.value);
```

Comment:
John, the last paragraph sums up what I meant by what i would propose depending on the answer to "why do we need to restict empty strings?".  But yeah letting thru and stopping is how i was just modeling the code in my head. but by letting thru i mean to use json.value, and stopping means dont use json.value which translates to use map.value. Basically what value we are letting thru to the server. Very obtuse way i think about things sometimes.

And yes to the question of map.value defaults to null thats the value we wind up sending to the server.

Andrew, immutable js will use the default value, which in this case is null when you dont provide a value, which generally means that the value is undefined. So what i am saying is basically that we dont need that check at all. Just use json.value. I am pretty sure Immutablejs will handle the coercion to null when json.value is undefined, but worth a tripple check.